var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -55,11 +56,14 @@ class Rust(Package):
         description='Install Rust source files'
     )
 
-    depends_on('cmake', type='build')
-    depends_on('python@:2.8', type='build')
+    depends_on('python@2.7:', type='build')
+    depends_on('python@2.7:2.8', when='@:1.42.0', type='build')
```

Comment:
This should probably be `@:1.43` since the current beta, which should not have the Python 3 fix, will become 1.43.

Reply:
Also, this probably shouldn't specify the patch version like you did with "1.42.0"? Spack version ranges confuse me sometimes, but you'll want this spec to match ALL releases of "1.43", even if they release a patch version of "1.43.1" later.